mypy/subtypes.py

```diff
@@ -143,10 +143,15 @@ def visit_instance(self, left: Instance) -> bool:
             rname = right.type.fullname()
             if not left.type.has_base(rname) and rname != 'builtins.object':
                 return False
-
             # Map left type to corresponding right instances.
             t = map_instance_to_supertype(left, right.type)
-
+            # Check that one of the types does not have type args or
+            # the number of type args is the same and
+            # each left type arg is acceptable in place of the matching right one
+            if not t.args or not right.args:
+                return True
+            if len(t.args) != len(right.args):
+                return False
```

Comment:
No, ``buitlins.tuple`` should always have one type argument, it represents a fallback ``Instance`` for comparisons with other ``Instance``s. For example, is ``Tuple[int, float]`` a subtype of ``Sequence[float]``, etc. I have found this code:
```python
fallback_item = join.join_type_list(items)
return TupleType(items, self.chk.named_generic_type('builtins.tuple', [fallback_item]))
```
This is what should ideally happen always.
Note that in ``typeanal.py`` types are not fully analysed in second pass, so that it is better to fix the tuple fallback in the third pass.